From 133afe7c998ed192f86ed2a8701d52c608404053 Mon Sep 17 00:00:00 2001
From: Rico Ganahl <rico.ganahl@bytesatwork.ch>
Date: Thu, 13 Apr 2023 08:47:31 +0200
Subject: [PATCH] bytesatwork: bawconfig: stm32mp1 ram fallback

Add fallback RAM for STM32MP1 on byteDEVKIT.

Signed-off-by: Rico Ganahl <rico.ganahl@bytesatwork.ch>
(cherry picked from commit 8fb2cc005e9d142dac53e567a14b21bd1a540083)
---
 board/bytesatwork/baw_config/baw_config_get.c | 21 +++++++++++++++++--
 1 file changed, 19 insertions(+), 2 deletions(-)

diff --git a/board/bytesatwork/baw_config/baw_config_get.c b/board/bytesatwork/baw_config/baw_config_get.c
index d2894e6cc4..4327f7f22f 100644
--- a/board/bytesatwork/baw_config/baw_config_get.c
+++ b/board/bytesatwork/baw_config/baw_config_get.c
@@ -40,6 +40,23 @@ void baw_config_get(struct baw_config *config)
 default_config:
 #endif
 
-	config->ram = M2_RAM_K4B2G1646EBIH9;	/* set default to legacy DDR3 */
-	printf("Error: no RAM configuration found, trying fallback configuration %u\n", config->ram);
+	if (IS_ENABLED(CONFIG_BAW_CONFIG_BUILTIN)) {
+		config->ram = BAW_CONFIG_BUILTIN_RAM;
+		if (IS_ENABLED(CONFIG_SPL_BUILD)) {
+			printf("Use built in RAM config: %u (%s)\n", config->ram,
+			       baw_config_get_ram_name(config->ram));
+
+			return;
+		}
+	} else {
+		if (IS_ENABLED(CONFIG_TARGET_BYTEENGINE_AM335X))
+			config->ram = M2_RAM_K4B2G1646EBIH9;	/* set default to legacy DDR3 */
+		else if (IS_ENABLED(CONFIG_TARGET_BYTEDEVKIT))
+			config->ram = M2_RAM_K4B4G1646DBIK0;
+	}
+
+	if (IS_ENABLED(CONFIG_SPL_BUILD)) {
+		printf("Error: no RAM configuration found, trying fallback config ");
+		printf("%u (%s)\n", config->ram, baw_config_get_ram_name(config->ram));
+	}
 }
-- 
2.39.2

